### PR TITLE
🐛 Tracker upgrade should use gvk from scheme

### PR DESCRIPTION
**Pull request by fabriziopandini**
This PR makes sure `tracker.Upgrade` uses the gvk from scheme for `AllowUnconditionalUpdate` and `AllowCreateOnUpdate` in case `TypeMeta` is not defined on the obj

Fixes: https://github.com/kubernetes-sigs/controller-runtime/issues/1212

### CODE REVIEW

File: pkg/client/fake/client.go [hunk 1]
```diff
@@ -114,11 +115,20 @@ func (t versionedTracker) Update(gvr schema.GroupVersionResource, obj runtime.Ob
 			field.ErrorList{field.Required(field.NewPath("metadata.name"), "name is required")})
 	}
 
+	gvk := obj.GetObjectKind().GroupVersionKind()
+	emptyGvk := schema.GroupVersionKind{}
+	if gvk == emptyGvk {
```
Comment (vincepri) [inline pkg/client/fake/client.go:120]:
```suggestion
	if gvk == schema.GroupVersionKind{} {
```

**Comment by detiber** [comment]
lgtm pending Vince's suggestion and fixing the PR title to appease the checks.

File: pkg/client/fake/client.go (hunk 1 above)
Comment (fabriziopandini) [reply pkg/client/fake/client.go:120]: @vincepri I have tested this syntax first, but it is failing to compile with "expected declaration, found oldObject"

Comment (vincepri) [reply pkg/client/fake/client.go:120]: Oh I think you can just do `gvk.Empty()`

**Review by vincepri — APPROVED** [review]
/approve

**Comment by vincepri** [comment]
@fabriziopandini Please fix the PR title

cc @DirectXMan12 Not sure why the check isn't running anymore 🤔

**Pull request merged**